.github/workflows/publish.yml

```diff
@@ -0,0 +1,356 @@
+
+      # If we're performing a nightly build and it fails acceptance tests then raise an issue.
+      - name: Alert on nightly build failure
+        uses: JasonEtco/create-an-issue@v2
+        if: ${{ failure() && github.event_name == 'schedule' }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          WORKFLOW_NAME: ${{ github.workflow }}
+          WORKFLOW_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+        with:
+          update_existing: true
+          filename: ./noir/.github/NIGHTLY_BUILD_FAILURE.md
+
+  build-windows:
+    needs: [build-barretenberg]
+    runs-on: windows-2022
+    env:
+      CROSS_CONFIG: ${{ github.workspace }}/.github/Cross.toml
+    strategy:
+      matrix:
+        target: [x86_64-pc-windows-msvc]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.tag || env.GITHUB_REF }}
+          path: ./noir
+
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Download artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: libbarretenberg-wasm32
+          path: ${{ github.workspace }}/libbarretenberg-wasm32
+
+      - name: Setup toolchain
+        uses: dtolnay/rust-toolchain@1.66.0
+        with:
+          targets: ${{ matrix.target }}
+
+      - name: Build environment and Compile
+        working-directory: ./noir
+        env:
+          BARRETENBERG_BIN_DIR: ${{ github.workspace }}/libbarretenberg-wasm32/bin
+        run: |
+          cargo build --package nargo_cli --release --target ${{ matrix.target }} --no-default-features --features plonk_bn254_wasm
+
+      - name: Package artifacts
+        working-directory: ./noir
+        run: |
+          mkdir dist
+          cp ./target/${{ matrix.target }}/release/nargo.exe ./dist/nargo.exe
+          7z a -tzip nargo-${{ matrix.target }}.zip ./dist/*
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: nargo-${{ matrix.target }}
+          path: ./noir/dist/*
+          retention-days: 3
+
+      - name: Test built artifact
+        working-directory: ./noir
+        shell: powershell
+        run: |
+          cp ./target/${{ matrix.target }}/release/nargo.exe ~/.cargo/bin/
+
+          cd release-tests
+          yarn install
+          yarn test
+
+      - name: Upload binaries to release tag
+        uses: svenstaro/upload-release-action@v2
+        if: ${{ inputs.publish || github.event_name == 'schedule' }}
+        with:
+          repo_name: noir-lang/noir
+          repo_token: ${{ secrets.NOIR_REPO_TOKEN }}
```

Comment:
Actually, we can remove this now right @phated?

Reply:
Yes, you can also remove the repo_name since you are running it within noir-lang/noir

Reply:
I don't think that we can remove it entirely due to permissions.

Reply:
As commented above, I think GITHUB_TOKEN works here. As with all our workflows, we should also be adding the `permissions` top-level section to the yaml files to know we are requesting the right perms, and only the right perms